modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/DefaultComponentParameterAst.java

```diff
@@ -138,7 +139,7 @@ protected void defaultVisit(MetadataType metadataType) {
             } else if (!NOT_SUPPORTED.equals(getModel().getExpressionSupport())) {
               if (isExpression(rawValue)) {
                 // For complex types that may be the result of an expression, just return the expression
-                expression.set(rawValue);
+                expression.set(extractExpression(rawValue).get());
```

Comment:
check for optional presense before setting: `extractExpression(rawValue).ifPresent(expression::set)`